plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSourceProvider.java

```diff
@@ -1149,19 +1091,14 @@ else if (column.isFileModifiedTimeColumn()) {
                     constantPopulatingPageSourceBuilder.addConstantColumn(nativeValueToBlock(FILE_MODIFIED_TIME.getType(), packDateTimeWithZone(fileModifiedTime.orElseThrow(), UTC_KEY)));
                 }
                 // For delete
-                else if (column.isRowPositionColumn()) {
+                else if (column.isMergeRowIdColumn() || column.isRowPositionColumn()) {
```

Comment:
i find this mixing a bit misleading to read.
(unfortunately i don't have a better suggestion)